spec/formatting.md

```diff
@@ -12,14 +12,15 @@ an appropriate error MUST be emitted and a _fallback value_ MAY be used as the f
 
 Formatting of a _message_ is defined by the following operations:
 
-- **_Expression and Markup Resolution_** determines the value of an _expression_ or _markup_,
+- **_Expression and Markup Resolution_** determines the resolved value of an _expression_ or _markup_,
   with reference to the current _formatting context_.
   This can include multiple steps,
   such as looking up the value of a variable and calling formatting functions.
   The form of the resolved value is implementation defined and the
   value might not be evaluated or formatted yet.
-  However, it needs to be "formattable", i.e. it contains everything required
-  by the eventual formatting.
+  However, if it is the resolved value of an expression with a function
```

Comment:
Why include the mention "with a function" here? Also, it's not really our functions that are "formatting" or "selection" functions, it's the resolved values that may or may not support either, or both.

Reply:
I agree with @eemeli. What's more, this is overly specific to expression and forgets about markup. Perhaps instead of trying to list everything separately we could say:

```suggestion
However, it needs to be "formattable", that is, it contains everything required
during later steps in the formatting process (such as selection, formatting, or output
of the _expression_ or _markup_.
```

Note that I was tempted to add "... as a string" at the end of the above, but `formatToParts` behavior shouldn't be omitted.

Reply:
> I'd recommend changing almost all usages to the specific terms:
> 
>     * "resolved expression value"
> 
>     * "resolved variable value"
> 
>     * "resolved function value", and so on

This is what #645 addresses, not using quite the same terms, but I encourage you to comment there (at some point) if you want to suggest changes.

Reply:
Thanks for the pointer, I'll look that over.

Reply:
We're aware of the post-45 work.

> The resolved value for a an expression does not have to be "formattable" if the expression uses a function that is not a formatting function.

Although I somewhat agree with you, I have always read "formatting" and "formattable" in this section to mean the processing found in this section--which include resolution, selection, and actual formatting. Formatting a message does not just mean making the output string.

Reply:
Since we strongly distinguish between formatting functions and selection functions, I think it very likely that people will misunderstand that "formatting" means "formatting and selection" in this paragraph.

That being said, this could wait for v46.